### DISCUSSION TITLE
기초 프로젝트 1주차 팀회고

### DISCUSSION THREAD

**Discussion by sulyoungoh** [codeit-bootcamp-frontend/3-Weekly-Mission#243] [회고]
# 왜 회고를 해야 하나요?
바쁘고 치열하게, 또 잘 버티고 있는 스프린트 3기 여러분!
정신없이 프로젝트를 하다 보면 현재 우리 팀의 방향성과 목표를 잊어버리기 쉬워요.
매주 팀원들과 일주일을 회고하며 프로젝트를 위한 원동력이 되었으면 좋겠어요.

# 회고에 대한 궁금증
회고와 일기의 차이는 무엇인가요?
- 일기는 일상에서 있었던 일들을 의식의 흐름대로 편하게 적는 것이라면,
회고는 함께 보고 듣고 경험한 사건과 현상들을 곱씹어 보고, 그중에 특히 되짚고 싶은 것들을 **'왜' 와 '어떻게'에 집중해서 기록**하는 것입니다.

# 회고 방법
<img width="636" alt="스크린샷 2024-01-17 오전 10 43 31" src="https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/assets/152246380/c1e8d800-0ef4-4f68-9b6c-076de80d7c3d">

회고는 **“KPT 회고”** 방식으로 진행해볼거예요. 아래 방법대로 회고해주시면 됩니다.

* 팀원들과 모여 그 주의 팀 프로젝트 활동을 돌아보며, '좋았던 점(Keep) / 아쉬웠던 점(Problem) / 시도해볼 것(Try)'으로 항목을 나누어 회고해주세요.
   - Keep : 현재 우리팀이 잘 하고 있어서 계속 유지하면 좋을 것 같은 부분
   - Problem : 현재 우리팀이 가지고 있는 문제
   - Try : Problem에 적은 것을 해결하기 위해 우리 팀이 다음에 시도해볼만한 것 (혹은 Keep 중에서도 더 개선하거나 시도해보고 싶은 것)
* [KPT 회고 피그마 템플릿](<https://www.figma.com/file/0NHIhlM9vhYZDrtZa7nJn8/%ED%9A%8C%EA%B3%A0-%ED%85%9C%ED%94%8C%EB%A6%BF?type=whiteboard&node-id=0-1&t=st0mINMyPQkSLXDd-0>)을 활용해서, 팀 회고를 진행하셔도 됩니다. 
   - 단 템플릿 사용시, 팀별로 복제해서 사용해주세요. 꼭 템플릿을 사용하지 않아도 되니, 팀별로 편한 방식으로 회고를 진행해주세요.
* **이번주 일요일까지, 팀 회고한 내용을 정리하여 댓글로 올려주세요.** 
(댓글에 예시 회고를 올려드릴테니, 참고해주세요. 피그마 템플릿으로 진행하셨다면 캡쳐본을 올려주셔도 돼요.)

## 참고글

- [KPT 회고란 무엇인가?](<https://brunch.co.kr/@jinha0802/35>)
- [팀문화의 탄생](<https://techblog.woowahan.com/2677/>)

**Comment by sulyoungoh** [comment]
# 🍀 0팀 (예시)

**Keep**
- 미팅 시간 약속을 모두 잘 지켜준다.
- 프로젝트 과정에서 기술 고민을 함께 하는 것이 서로의 학습에 도움이 된다.
- 새로운 제안을 했을 때, 모두 개방적으로 의견을 들어주어서 고맙다.
- 위워크에 모여서 하다보니, 프로젝트 진행이 빠르게 잘 이루어진다.

**Problem**
- 커밋 메시지가 명시적이지 않아서, 서로 어떤 작업을 했는지 파악하기 힘들었다.
- 급하게 코드를 작성하느라, 폴더 구조가 이상하다.
- 서로를 너무 배려하느라, 의견을 더 많이 나누지 못한다.

**Try**
- 협업을 위해, 커밋 메시지를 더 구체적으로 작성한다. (커밋 메시지 규칙 정하기)
- 폴더 구조를 다시 논의하여 폴더 구조 규칙을 만들고, 규칙에 맞추어 파일을 배치한다.
- 각자 하루에 한 번 이상 의견을 낸다.

**Comment by DanteSnow** [comment]
### ⭐️**5팀**

### 💡**Keep**

- 팀원들끼리 똘똘 뭉쳐서 서로 챙겨주고 소통하는 분위기가 형성되었다. 팀 프로젝트가 끝날 때 까지 유지되었으면 좋겠다.
- 5명이서 머리를 맞대고 활발한 소통을 하니, 문제가 생겨도 금방 해결을 할 수 있었다.
- 팀 프로젝트를 하면서 깃허브에 대한 이해도가 올라갔다.
- 팀 프로젝트를 진행하며 소통하는 방법을 많이 배웠다.
- 깃 브랜치, upstream, origin에 대한 지식이 헷갈렸는데, 팀 프로젝트를 하면서 전보다는 이해가 더 잘 되었다.
- 새로 알게된 기능들을 주저하지 않고 사용함으로 내 것으로 만들었다.

### ❓**Problem**

- 팀 프로젝트를 접해본 경험이 부족했기 때문에 초기 세팅(툴 설치, 깃허브 세팅)과 일정 관리 부분이 미숙했다.
- 회의 시간이 길어지다보니 집중력이 낮아졌다. 그래서 의견을 적극적으로 내지 못했다.
- 일정을 짜는 것에 부족함이 있었어서 아쉬웠다.
- 깃허브의 기능 사용에 대해서 익숙하지 않다.

### 🏃**Try**

- 팀 프로젝트를 진행하며 겪는 문제들과 그 문제들을 어떻게 해결했는지를 꼼꼼하게 회고하며 경험을 쌓아갈 것이다.
- 회의 시간이 길어지면 쉬는 시간을 조금씩 가지는 것이 좋을 것 같다.
- 좋은 의견이 있으면 적극적으로 제안할 것이다.
- 추가 변수가 생길 시를 대비하여 꼭 여유 시간을 챙길 것이다.
- 깃허브 기능들에 대해서 더 알아보고, 익숙하지 않아도 매일 사용할 것이다.

**Comment by jihee1103** [comment]
## 8팀
### Keep
- 궁금한것이 생기면 이해가 될 때 까지 학습하는 끈기
- 게더에 모여서 즉시 궁금한점을 해결하는것이 좋다
- 적극적으로 서로 알려주며 진행을 해서 깃,코드에 대한 이해도가 높아진 것 같다.
- 모두가 자유롭게 의견을 낼 수 있는 분위기
- 자료를 사전에 열심히 조사하는 습관.

### Problem
- 이해되지 않거나 오류가 생기는 경우 쉽게 지치는 것
- 작업내용/코드에 대한 공유가 부족한 것 같다.
- 파일이 너무 많아서 적응하기가 힘들다
- 코드에 대한 기록을 꼼꼼히 하지 못하고 지금은 구현을 위해 일단 코드를 짜기만 하고 있다.

### Try
- 스트레스 너무 깊게 받지 않고, 해결되지 않는 오류는 없다! 는 마인드로 임한다. 보통 내 실수에서 일어난 오류가 대부분이고, 해결 되지 않는 것은 없다.
- PR에 작업내용을 필수적으로 적기
- 모르는게 있으면 바로바로 물어보기
- 내가 짠 코드에 대한 리뷰를 하며 최대한 틈틈이 기록하기

**Comment by Jnnsu** [comment]
##  🍀 3팀

### 💡**Keep**

- 의견을 무작정 주장하지 않고 서로 배려하며 제안하듯 말하는 점이 좋았습니다.
- 실행해야 되는 것들, 기억해두어야 하는 것들 메모로 항상 남겨두는 것도 배울 점이라 생각합니다.
- 코드 작성하는 부분에서 다양성이 있어서 제가 작성하는 코드의 단점과 장점을 좀 더 쉽게 알 수 있었던 거 같습니다.
- 문제가 생겼을 때 해결할 수 있는 사람이 있다면 망설임없이 도와주는게 좋았던 것 같습니다. 덕분에 이해가 가지 않거나 이상하다 하는 부분들을 이야기할 수 있었습니다.
- 라이브쉐어로 공통 컴포넌트들을 함께 만들면서 다른 사람은 같은 상황에 어떻게 코드를 작성하는지 바로바로 물어볼 수 있어서 좋았습니다.
- 플젝을 시작하기전에는 막연한 두러움과 괜한 걱정들이 많이있었는데 혼자서 할때보다 막히는게 있었다면 여러명이서 다각적으로 보고 접근방식들도 다양해져서 문제들이 잘 풀렸고 무엇보다 든든한게 컸던것 같습니다.

### ❓**Problem**

- 모르는 부분에 대해 공유하고 질문하는 게 부족하다고 생각됩니다.
- 코드를 작성하는 부분에서 디테일과 일관성을 더 높이면 좋겠다고 생각합니다.
- 막히는 부분에서 고민하는 시간이 긴 거 같습니다. 사소한 거라도 얘기를 나누면서 해결법을 찾거나, 해결법이 생각이 나지 않는다면 원래 목적을 간소화해서라도 일단은 동작이 되는 코드를 작성하는게 효율적으로 보입니다.
- 이슈에 갇혀버렸을 때, 그 문제를 온전히 해결하고 쉬는 것도 좋지만 너무 오래 고민한 경우 조금 쉬었다 하는 것도 방법이라 생각합니다. 쉬면서 각자 문제에 대해 검색해보거나 할 수 있을 것 같습니다. 겸사겸사 검색능력도 키우면 좋을 것 같아요.
- 다른 팀의 진행 상황은 잘 모르겠지만 예상치 못한 에러들이 생겨서 예상보다 진도가 늦어지는 것 같아 다음 주에 바쁘게 보내야 할 것 같습니다.
- 플레이타임을 절절하게 분배하지 못한것 같아서 조금 아쉬웠습니다. 사람마다 체력이 다르고 집중할수있는 시간이 다른만큼 그에 대한 고려도 조금더 고려해봤다면 지금보다 조금더 나은 우리가 될수 있을것 같다는 생각이 듭니다.

### 🏃**Try**

- 커뮤니티를 좀 더 잘 활용해서 서로의 상황을 좀 더 공유해야 될 것 같습니다.
- 더 디테일한 컨벤션 설정이 추가적으로 필요할 것 같습니다.
- 컴포넌트를 만드는 것에 있어 조금 더 능숙해져서 이후에 컴포넌트 사용에 있어 문제가 생기지 않도록 노력해야 될거 같습니다.
- 추가로 제공된 리액트 강의를 틈틈히 듣고 있습니다.. 더 잘 다루기 위해 이것저것 찾아보겠습니다.
- 각자 맡은 페이지를 만들면서 에러가 발생했을 때 다같이 해결하고 넘어가는 방식이 좋을 것 같습니다.
- 혼자서 하는 플젝이 아닌만큼 혼자서 다 하겠다는 생각말고 멀리 보는것도 중요하지만 현재 바로 앞에 집중하고 얘기를 나누어서 함께 나아가는 방향으로 해보자!

**Comment by soyxxn** [comment]
### **🩷2팀🩷**
-----------------------------

🥰 **Keep**
- 시간 약속을 다들 잘 지켜준다
- 모르거나 부족한 부분을 서로 내 문제처럼 도와주고 적극적으로 해결해주려 한다
- 다들 참여율이 높고 열의가 보여서 덩달아 열심히 하게 된다
- 소통이 잘 된다
- 그렇기 때문에 빨리 수정할 수가 있다!

🤔 **Problem**
- 커밋 메시지를 조금 더 명확하게 작성하고, 기능별로 나눠 커밋할 필요가 있을 것 같다
- 의견이 부족한 거 같다
- 한 부분에서 막히면 쉽게 집중력이 흐려진다
- 도움이 필요해도 먼저 도움 요청을 하지 못하고 혼자 끙끙거렸다
- 아는 게 적다고 생각해서 회의에 적극적으로 참여하지 못했다
- 팀 프로젝트가 처음이라 아직까지는 낯설고 쉽지 않은 거 같다
- 깃허브 사용이 아직도 어렵다

🤓 **Try**
- 모르는 부분에 있어서 더 적극적으로 물어보고 꼼꼼히 알아볼 시간이 필요한 것 같다!
- 도움이 필요할 때 지체하지 않고 조언을 구하는 게 좋을 거 같다! 미루는게 오히려 팀에 손해라는 생각이 들었다
- 막히는 부분이 있으면 멘토님, 팀원분들 그리고 구글링을 적극 활용하자!
- 쓸 수 있는 부분을 적극적으로 활용하면 좋을 것 같다! 다같이 기록을 잘 했으면 좋겠다!
- 기능 구현이 금방 끝날 거 같아서 추가할 기능 생각나면 공유하자!
- 하고 있는 작업에 대해 그때그때 말하기!

**Comment by yuyoni** [comment]
# ✨ 1팀

## 👍 Keep

- 프로젝트 시작 시 전체적인 흐름과 폴더 구조, 설정 등을 최대한 맞추고 시작한 점이 좋았다.  협업 시 시간이 가장 많이 투자되는 부분이 이러한 규칙들을 정하지 않고 생기는 문제들이라고 생각하고 있어, 조금 더 원활한 팀 프로젝트가 가능할 것 같다.
- 서로 오류나 궁금한 것이 있으면 같이 공유하고 해결하는 부분과 서로의 일정을 공유하는 부분도 좋은 것 같다.
- 어려워보이는것도 일단 해보고 보자 라는 기조덕에 새로운것들을 많이 경험해서 좋았다.
- 서로 간 피드백도 활발했고 정보공유도 적극적이라 성장에 도움이 많이 된다.
- 방향성을 잘 잡고 체계적으로 구조화하여 계획했기 때문에 순조롭게 진행이 되는 것 같다.

## 💭 Problem

- 네이밍 컨벤션에 아직 미숙하다보니 통일성이 떨어지는 느낌이 있는 것 같다.
- 아직 초반이라 폴더 구조를 한눈에 알아보기 조금 어려운 느낌도 있다.
- 코어 타임 없이 각자 작업을 해서 부재중인 경우 꼭 얘기해야하는 사항이나 지금 바로 수정해야하는 것들을 해결할 수 없어서 불편했다.
- 처음에 정한 원칙이 편의상 변경하려는 시도들이 있었다. 원칙은 절대 변경되어선 안되는 것이 아니지만, 그 목적이 분명이 분명하고 중요했던 원칙이 쉽게 변경되는 일은 없었으면 좋겠다.
- 코드 쓰는방식이 제각각이라 아직까진 같이 작업한다기보단 각자 따로 작업을 한다는 느낌이 조금 든다.

## 💡 Try

- 조금 더 컨벤션을 구체적으로 정해서 개발에 집중할 수 있으면서도 통일성 있는 코드를 작성할 수 있도록 하고 싶다!
- 계속해서 서로의 코드 리뷰를 통해 네이밍 컨벤션, 스타일드 컴포넌트 사용방식 등을 맞추어 나가면 좋을 것 같다.
- 지금은 페이지 레이아읏과 기능 구현이 우선이지만, 모든 팀원이 각자 작업을 마무리했을 때 폴더 구조를 보기 좋게 설계해 보는 것도 좋을 것 같다.
- 다 같이 하는 시간,코어 타임을 정해서 게더타운이나 디스코드 등을 활용해 함께 작업하고 바로바로 확인할 수 있는 시간이 있으면 더 좋을 것 같다.
- 협업에 중점을 두는만큼 각자 구현은 하되, 어떤식으로 구현하였고, 어떤 기능들을 썼는지 리뷰하는 시간이 있으면 좋을 것 같다! (PR에 익숙하지 않기 때문에 코드를 보는 것만으론 의도를 파악하기 어려울 때가 있음)

**Comment by SiWooJinSeok** [comment]
## 6팀

## :thumbsup:  **Keep**

- 모두 포기하지 않고 열심히 한다.
- 모두 많은 것을 시도하는 것을 즐긴다.
- Slack을 통해서 서로의 상황을 공유한다.
- 서로 코드 리뷰를 해주고 잘하거나 배운 점도 같이 작성해준다.

## :eyes:  **Problem**

- 문서 작성이 어렵다.
- 새로운 라이브러리들이 추가 되어 배우는 시간이 걸린다.
- 노션, 깃허브, Slack등 아직 사용하는 것이 익숙하지 않다. 

## :raised_hands:   **Try**

- 서로 맡은 부분에서 최선을 다하고, 서로 도와갈 것이다.
- 어려운 부분은 서로 도와가며 도움을 요청해보기

**Comment by ohddang** [comment]
## 7팀

## 🌼keep

- 서로의 의견 수용을 잘해준다.
- 프로젝트 시작 시 정한 규칙를 고집하지 않고, 프로젝트를 진행하면서 짧은 주기로 피드백을 나누고 필요한 부분을 변경한다.
- 의사소통이 굉장히 잘되는 편이고, 뭘 물어보거나 의견을 제시했을 때 답변이 빠르다.
- 기능 개발 속도가 빠르다.
- 팀플을 즐기면서 한다.
- 코드 리뷰를 통해 서로의 코딩 스타일을 확인하며, **미처 발견하지 못했던 실수를 보완한다.**

## 😯problem

- 각자 개발을 하고 그 이후에 피드백이 부족한 거 같다.
- 지금 다른 팀원이 무얼 하고 있는지 정확하게 알기 어렵다.

## ✨try

- 중요한 기능이나 개발 내용이 많을 경우 자신이 개발한 내용에 대해서 발표하는 시간을 가졌으면 함
- 코드를 이해할 수 있도록 주석을 달아서 표시해(Todo, Info, Fix)
- Jira 이슈를 좀 더 세분화해서 등록하고 진행할 수 있도록 할 것
- 그날 뭐 했는지 디스코드에 남기기

**Comment by jiyong0106** [comment]
### 4팀
Keep

- 적극적인 팀미팅 소통.
- 배웠지만 사용하지 않았던 기술에 대해 도전하는 것을 긍정적으로 받아들인다.
- 매일 전날 구현했던 부분을 서로 공유하며 진행 상황을 알려준다.
- 각자의 코딩 스타일을 인정하고 사용하는 방법에 대해 주석을 자세히 적어준다
- 맡은 부분에 대해서 책임지고 완성하려는 부분이 멋있다.
- 고민하던 내용을 이야기 하는 것에 있어서 주저하지 않는다.

Problem

- 다들 프로젝트가 처음이라 컨벤션을 정하는 것에 있어서 시간이 오래 걸렸다.
- 프로젝트 회의 내용 상남자식 노션 정리
- 폴더 구조 및  네이밍에 대한 어려움이 있다.

Try

- 프로젝트에 추가할 수 있는 다른  기능을 찾아보고 소통하고 공유하기
- 폴더, 파일, 코드 네이밍을 통일해보기
- pull request 를 좀 더 활용해보기
